packages/esm-patient-registration-app/src/patient-registration/patient-registration-utils.ts

```diff
@@ -192,11 +192,59 @@ export function getPatientIdentifiersFromFhirPatient(patient: fhir.Patient): Arr
   });
 }
 
-export function getPhonePersonAttributeValueFromFhirPatient(patient: fhir.Patient) {
+export async function getIdentifierFieldValuesFromFhirPatient(
+  patient: fhir.Patient,
+  identifierConfig,
+): Promise<{ [identifierFieldName: string]: PatientIdentifierValue }> {
+  const identifiers: FormValues['identifiers'] = {};
+  const promises: Promise<void>[] = [];
+
+  for (const identifier of patient.identifier) {
+    for (const config of identifierConfig) {
```

Comment:
Rather than two nested for loops, it seems like it would make sense to loop through the `identifierConfig` once and create an object that keyed from `fhirIdentifierSystem` to the `config` object associated with it. That way, we loop through it once and the lookup from identifier -> identifierConfig is an O(1) operation.